js/src/views/Addresses/addresses.js

```diff
@@ -72,27 +72,40 @@ class Addresses extends Component {
   }
 
   render () {
-    const { balances, contacts, hasContacts } = this.props;
-    const { searchValues, sortOrder } = this.state;
-
     return (
       <div>
         { this.renderActionbar() }
         { this.renderAddAddress() }
         <Page>
-          <List
-            link='address'
-            search={ searchValues }
-            accounts={ contacts }
-            balances={ balances }
-            empty={ !hasContacts }
-            order={ sortOrder }
-            handleAddSearchToken={ this.onAddSearchToken } />
+          { this.renderAccountsList() }
         </Page>
       </div>
     );
   }
 
+  renderAccountsList () {
+    const { balances, contacts, hasContacts } = this.props;
+    const { searchValues, sortOrder } = this.state;
+
+    if (hasContacts && Object.keys(balances).length === 0) {
+      return (
+        <Loading size={ 3 } />
```

Comment:
Worried about the loading sizes everywhere - should pick one size and just stick with it throughout. (Only exceptions may be whole pages where everything is blank)